### ISSUE TITLE
validation fails on developer IDs with uppercase 

### ISSUE THREAD

**Issue by A6GibKm** [ximion/appstream#617]
The following:

```xml
  <developer id="io.github.Uppercase">
    <name>Hello</name>
  </developer>
```

produces a warning:

```
$ appstreamcli validate --explain app.metainfo.xml 
W: app.drey.Test:6: developer-id-invalid io.github.Uppercase
   The developer-ID is invalid. It should be an rDNS string identifying the developer, or a
   Fediverse handle. It must also only contain lowercase ASCII letters, numbers and punctuation.
```

The docs at https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-developer do not mention any requirement on lowercase vs uppercase.

Either the docs are missing info or the linter is not doing the right thing.

**Comment by A6GibKm** [comment]
Note that `@User@example.org` would also fail. Can a Fediverse handle be case sensitive? If so it makes sense to allow uppercase in those.

**Comment by ximion** [comment]
I agree, sorry for the delayed response! (I am very preoccupied at the moment)
We could make this an info hint, like with component-IDs, but TBH I think the better thing to do is to just drop the case check, especially since it does not follow the spec in that regard.

I faintly remember that we initially wanted to allow only lowercase, then dropped that, and it looks like I did not adjust the check (it was supposed to test only the domain part of the rDNS for case, not the whole string...).
Maybe I will get to it today, otherwise some time during the week :-)

**Comment by A6GibKm** [comment]
Hello, is there any update to this? We are using the app id in some apps like e.g. Bustle, and since the GNOME runtime now has a new enough appstream the app fails at CI.

```xml
  <developer id="org.freedesktop.Bustle">
    <name>The Bustle developers</name>
  </developer>
```
